commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/HostFileNameParser.java

```diff
@@ -357,4 +369,21 @@ public FileName parseUri(final VfsComponentContext context, final FileName base,
         return new GenericFileName(auth.scheme, auth.hostName, auth.port, defaultPort, auth.userName, auth.password,
                 path, fileType);
     }
+
+    private static boolean isIPv6HostHeadingChar(final char ch) {
+        return ch == '[';
+    }
+
+    private static boolean isHostNameTerminatingChar(final char ch, final boolean isIPv6Host) {
+        if (isIPv6Host) {
+            return ch == ']';
+        }
+
```

Comment:
Hi @fyudanov 

Thank you for this PR! 😄 

The expression below does not look right to me or the method is misnamed, but the code sure could use some comments 😉 

Below, we have the test for the last char of a host name:
```
 return ch == '/' || ch == ';' || ch == '?' || ch == ':' || ch == '@' || ch == '&' || ch == '=' || ch == '+'
                || ch == '$' || ch == ',';
```
Reading https://datatracker.ietf.org/doc/html/rfc3986#page-49:
```
host          = IP-literal / IPv4address / reg-name
...
reg-name      = *( unreserved / pct-encoded / sub-delims )
...
unreserved    = ALPHA / DIGIT / "-" / "." / "_" / "~"
sub-delims    = "!" / "$" / "&" / "'" / "(" / ")"
                 / "*" / "+" / "," / ";" / "="
```
For example, shouldn't `~` be in the test. Also the method needs comments if you intend to also test for a separator char with rest of the path like `/` and `?`. Or am I missing something?

At least this PR needs more testing with host names and URI that contain all the these special characters.

Reply:
Hi @garydgregory 

I agree that the char list from
```
ch == '/' || ch == ';' || ch == '?' || ch == ':' || ch == '@' || ch == '&' || ch == '=' || ch == '+'
                || ch == '$' || ch == ',';
```
looks a bit inconsistent with RFC (at first site at least), but it wasn't introduced or changed in this PR, I've just moved it to the separate method.

Not sure, if we want to revisit it in the scope of the current change, because the IPv6 host parsing algorithm doesn't actually depend on this list :)

I could add few more tests with chars like `~`, if it would be sufficient, but to me, this would cover previously existing IPv4 cases rather than the current change.

Reply:
More tests would be great.

Reply:
Ok, I've added few new test cases. Those that are commented out are the cases that doesn't work like I would expect according to RFC.

However, I've ensured the behavior on the tests is exactly the same as it was before this PR, so I'm not sure we want to cover missed cases here.